lib/src/clojure_lsp/feature/rename.clj

```diff
@@ -192,62 +191,60 @@
     :else
     (mapv (partial rename-other replacement db) references)))
 
-(defn ^:private rename-status
-  [element definition references source-paths client-capabilities]
-  (cond
-    (empty? references)
-    {:error {:message "Can't rename, no other references found."
-             :code :invalid-params}}
-
-    (and (= :namespace-definitions (:bucket definition))
-         (not= :namespace-alias (:bucket element))
-         (seq source-paths)
-         (not-any? #(string/starts-with? (:filename definition) %) source-paths))
-    {:error {:code :invalid-params
-             :message "Can't rename namespace, invalid source-paths. Are project :source-paths configured correctly?"}}
-
-    (and (= :namespace-definitions (:bucket definition))
-         (not= :namespace-alias (:bucket element))
-         (not (get-in client-capabilities [:workspace :workspace-edit :document-changes])))
-    {:error {:code :invalid-params
-             :message "Can't rename namespace, client does not support file renames."}}
-
-    (and (contains? #{:keyword-definitions :keyword-usages} (:bucket definition))
-         (not (:ns definition)))
-    {:error {:code :invalid-params
-             :message "Can't rename, only namespaced keywords can be renamed."}}
-
-    :else
-    {:result :success}))
+(defn ^:private rename-status [db element]
+  (let [references (q/find-references db element true)
+        definition (q/find-definition db element)
+        client-capabilities (:client-capabilities db)
+        source-paths (settings/get db [:source-paths])
+        source-path (shared/uri->source-path (:uri definition) source-paths)]
```

Comment:
Calculating the `definition`, `references` and `source-path` in `rename-status` lets us remove some duplication between `prepare-rename` and `rename-element`

Reply:
good catch